### ISSUE TITLE
[Bug]: (Photos' names out of order)

### ISSUE THREAD

**Issue by shinkiro25** [KJHJason/Cultured-Downloader#65]
### Which program are you seeing this problem on?

Cultured Downloader (py)

### Program Version

4.2.3

### Steps to reproduce the bug

1. Trying to download all posts from an artist
2. 
3. 
4.

### Expected behaviour

Ordered photos in the downloaded folders

### Relevant log output (If any)

I see unordered names like
""I:\FTP\Master18\[1450215] 2022 8 29\4f347288-0128-42b5-97a1-078f6ef55188.png"

and ""I:\FTP\Master18\[1450215] 2022 8 29\4d5920b6-2b32-43cb-9431-81a50ded02f3.png""

**Comment by KJHJason** [comment]
If it's downloaded from Fantia, Fantia uses hexadecimal for their images as seen below.
![image](https://github.com/KJHJason/Cultured-Downloader/assets/73630142/49808282-aee3-442b-bd07-fe94c73c67b2)

Though other services like Pixiv Fanbox also do this so it's not a bug but more so the program's logic which downloads the files using filenames according to its URL.

**Comment by KJHJason** [comment]
I could make this a feature request if you would like

**Comment by shinkiro25** [comment]
I'd love if it can be added as a feature.

This extension https://chrome.google.com/webstore/detail/fantia-downloader/cfkejojncmolgahkabegcdbcbmnghhcp
can already download them with proper ordered numbers in the file names, but it only work for one post. I don't know how they're pulling the numbers from Fantia.

**Comment by shinkiro25** [comment]
@KJHJason Hi Jason, any update on this feature? I'd be immensly grateful if it gets implemented.

**Comment by shinkiro25** [comment]
My apologies, I didn't know you had such hardships. The best of luck with your assignments and tests.

**Comment by shinkiro25** [comment]
@KJHJason Hi Jason, I hope your assingments and tests went well :) 

Any update on the new version?

**Comment by KJHJason** [comment]
I'm currently working on the GUI which will take some time before I start working on this issue.
However, my internship is starting next week so my progress will slow down but eventually, it will be done!

**Comment by shinkiro25** [comment]
Thank you, I hope this issue will be fixed :) And good luck with your internship as well.

**Comment by shinkiro25** [comment]
Hi again, has the issue been fixed?

**Comment by KJHJason** [comment]
Hello @shinkiro25 , I have implemented the logic and it will be in the upcoming V5.0.0 of Cultured Downloader!
I am targeting to release Cultured Downloader V5.0.0 by 18 May 2024 or earlier. So, I will be closing this issue as it's finished.

Last but not least, I have attached some screenshots of it in action!

![Image](https://github.com/KJHJason/Cultured-Downloader/assets/73630142/9d6af793-643b-41ed-a32b-6e62299e564e)

![Image](https://github.com/KJHJason/Cultured-Downloader/assets/73630142/e83f1669-bfb4-4865-ad71-286504bf1260)
